freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/conversion/UserMetadataConverter.java

```diff
@@ -0,0 +1,124 @@
+            if (protoMetadata.isPresent()) {
+                userMetadata = Optional.of(new UserMetadata(jsonUserMetadata.get().getCrn(), protoMetadata.get().getWorkloadCredentialsVersion()));
+            }
+        }
+
+        return userMetadata;
+    }
+
+    public String toUserMetadataJson(UserMetadata userMetadata) {
+        requireNonNull(userMetadata);
+        return toUserMetadataJson(userMetadata.getCrn(), userMetadata.getWorkloadCredentialsVersion());
+    }
+
+    public String toUserMetadataJson(String crn, long workloadCredentialsVersion) {
+        checkArgument(!Strings.isNullOrEmpty(crn));
+
+        // Store the metadata in a protobuf to make it easy to evolve the metadata model.
+        UserManagementProto.WorkloadUserSyncActorMetadata protoMetadata = UserManagementProto.WorkloadUserSyncActorMetadata.newBuilder()
+                .setWorkloadCredentialsVersion(workloadCredentialsVersion)
+                .build();
+        String protoMetadataBase64Encoded = BaseEncoding.base64().encode(protoMetadata.toByteArray());
+
+        // Wrap the CRN and metadata in JSON to make it easy to search for an exact match of the CRN string delimited by quotes.
+        JsonUserMetadata jsonUserMetadata = new JsonUserMetadata(crn, protoMetadataBase64Encoded);
+        try {
+            return OBJECT_WRITER.writeValueAsString(jsonUserMetadata);
+        } catch (JsonProcessingException e) {
+            throw new IllegalStateException(String.format("Unable to serialize %s as JSON: ", jsonUserMetadata), e);
+        }
+    }
+
+    private Optional<JsonUserMetadata> getJsonUserMetadata(User user) {
+        Optional<JsonUserMetadata> jsonUserMetadata = Optional.empty();
+        try {
+            jsonUserMetadata = Optional.of(OBJECT_READER.readValue(user.getTitle()));
+        } catch (JsonProcessingException e) {
+            LOGGER.warn("Unable to parse metadata string {} for user with workload username {}", user.getTitle(), user.getUid(), e);
+        }
+        return jsonUserMetadata;
+    }
+
+    private boolean isValidActorCrn(String crnToValidate, String username) {
+        Optional<Crn> crn = Optional.empty();
+        try {
+            crn = Optional.of(Crn.safeFromString(crnToValidate));
+        } catch (CrnParseException e) {
+            LOGGER.warn("Unable to parse CRN {} in metadata for user with workload username {}", crnToValidate, username, e);
+        }
+
+        boolean validCrn = false;
+        if (crn.isPresent()) {
+            Crn crnVal = crn.get();
+            if (crnVal.getService() == Crn.Service.IAM &&
+                    (crnVal.getResourceType() == Crn.ResourceType.USER || crnVal.getResourceType() == Crn.ResourceType.MACHINE_USER)) {
+                validCrn = true;
+            } else {
+                LOGGER.warn("CRN {} in metadata for user with workload username {} is not a user or machine user CRN", crnToValidate, username);
+            }
+        }
```

Comment:
I think this way it's a bit more readable and shorter:
```
    private boolean isValidActorCrn(String crnToValidate, String username) {
        try {
            Crn crn = (Crn.safeFromString(crnToValidate));
            if (crn.getService() == Crn.Service.IAM &&
                    (crn.getResourceType() == Crn.ResourceType.USER || crn.getResourceType() == Crn.ResourceType.MACHINE_USER)) {
                return true;
            } else {
                LOGGER.warn("CRN {} in metadata for user with workload username {} is not a user or machine user CRN", crnToValidate, username);
                return false;
            }
        } catch (CrnParseException e) {
            LOGGER.warn("Unable to parse CRN {} in metadata for user with workload username {}", crnToValidate, username, e);
            return false;
        }
    }

Reply:
Done.